src/test/java/com/eighthlight/fabrial/test/http/HttpHeaderWriterSerializationTests.java

```diff
@@ -0,0 +1,47 @@
+package com.eighthlight.fabrial.test.http;
+
+import com.eighthlight.fabrial.http.HttpHeaderWriter;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
+
+public class HttpHeaderWriterSerializationTests {
+  @Test
+  void allowHEAD() throws IOException {
+    try (var os = new ByteArrayOutputStream()) {
+      var writer = new HttpHeaderWriter(os);
+      writer.writeFields(Map.of("Allow", "GET"));
+      assertThat(os.toString(), is("Allow: GET \r\n"));
+    }
+  }
+
+  @Test
+  void allowMultipleMethods() throws IOException {
+    try (var os = new ByteArrayOutputStream()) {
+      var writer = new HttpHeaderWriter(os);
+      writer.writeFields(Map.of("Allow", "GET,HEAD,OPTIONS"));
+      assertThat(os.toString(), is("Allow: GET,HEAD,OPTIONS \r\n"));
+    }
+  }
+
+  @Test
+  void writeMultipleFields() throws IOException {
+    try (var os = new ByteArrayOutputStream()) {
+      var writer = new HttpHeaderWriter(os);
+      writer.writeFields(Map.of(
+          "Allow", "GET,HEAD,OPTIONS",
+          "Content-Length", "0"));
+      var expectedAllow = "Allow: GET,HEAD,OPTIONS ";
+      var expectedContentLength = "Content-Length: 0 ";
+      var headerLines = Arrays.asList(os.toString().split("\r\n"));
+      assertThat(headerLines, containsInAnyOrder(expectedAllow, expectedContentLength));
```

Comment:
Small thing, I usually find it easier to read through unit tests when there's a blank lines between setup/action and between action/assertion. So here maybe have `writer.writeFields(...)` have a newline before and after.

Reply:
I prefer to refactor into expressive helper methods where it makes sense, but where that fails, I've also found newlines helpful in distinguishing between the different "phases".